megamek/src/megamek/common/MovePath.java

```diff
@@ -114,13 +114,18 @@ public int hashCode() {
 
     // is this move path being done using careful movement?
     private boolean careful = true;
+    private boolean gravityConcern = false;
+    private final float gravity;
 
     /**
      * Generates a new, empty, movement path object.
      */
     public MovePath(final Game game, final Entity entity) {
         this.setEntity(entity);
         this.setGame(game);
+        // Do we care about gravity when adding steps?
+        gravity = game.getPlanetaryConditions().getGravity();
+        gravityConcern = (gravity != 1.0F && game.getBoard().onGround() && !entity.isAirborne());
```

Comment:
You could make this a little more as well - the only time gravity becomes a concern for low gravity is when the entity's MP actually changes as a result of gravity.

The reason I say this is because gravity and entity move MP don't change, but addStep() is called very frequently, so if we can move at least some of the MP checks out of the addStep method, that will help with performance.

Reply:
Don't MPs change as soon as gravity is over 1.0?  Let me double-check...

Reply:
Actually, I guess we can just ask the entity if its MP are different from no-gravity MP...

Reply:
Done.